### PR TITLE
Redux react components

### PR DESCRIPTION
**Pull request by AbrahaKahsay**
# Project description
In this project, I have implemented the use Redux store to display books. I have also added some UI elements to your components and make them dispatch actions. 
## Requirements

- First of all - make sure that you keep state immutable in reducers!
- Secondly, change the default state in your books reducer from an empty array to an array with a few books. One of the goals is to display them in your React components.
- Import Redux Provider and your store in the main component of your React app.
- Connect your app to the Redux store.
- Make sure that list of books displays books from Redux store.
     - Use useSelector to consume the state.
- In your React component responsible for adding new books, add functionality for "Add new" button click:
     - Import Redux useDispatch.
     - Set the data inputs in the local React state (set title and author - NOTE: categories and comments are NOT part of this step).
     - Takes the inputs from the state and generate a unique id and add them to an object.
     - Once your new book object is ready to be submitted to Redux store, dispatch a corresponding action (import Action Creator that you created before and use it here).
- In your React component responsible for removing books, add a functionality for "Remove" button click:
       - Import Redux useDispatch.
       - Dispatch a corresponding action (import Action Creator that you created before and use it here).
- In your React component responsible for categories, add a functionality for "Check status" button click:
     - Import Redux useDispatch.
     - Dispatch a corresponding action (import Action Creator that you created before and use it here).
 ## Built with
- Major languages: HTML, CSS, JavaScript, React, Redux
- Technologies used: VS Code, Github, Linters, Firefox web-browser
- webpack
- babel

### CODE REVIEW

File: src/components/Categories.js [hunk 1]
```diff
@@ -1,6 +1,13 @@
...
-  <button type="button">Check status</button>
-);
+const Categories = () => {
+  const dispatch = useDispatch();
+  return (
+    <>
+      <button type="button" onClick={() => dispatch(checkStatus())}>Check status</button>
+    </>
+  );
+};
```
Comment (julie-ify) [inline src/components/Categories.js]: - [ ] After dispatching the action to check the status, you still need to fetch the category state and display that text in this component. Kindly do that.

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @AbrahaKahsay,

Good job so far!

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Categories.js (hunk 1 above)
Comment (AbrahaKahsay) [reply src/components/Categories.js]: @julie-ify Thanks for your valuable review. I have implemented the changes.

**Review by rustam-tolipov — APPROVED** [review]
## Hello @AbrahaKahsay 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
## Congratulations! 🎉 🎉 🎉 🎉 🎉 

Cheers and Happy coding!👏👏👏

Developer-512px-10

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**